dataland-backend/src/main/kotlin/org/dataland/datalandbackend/service/DataManager.kt

```diff
@@ -0,0 +1,156 @@
+import org.dataland.datalandbackend.model.DataMetaInformation
+import org.dataland.datalandbackend.model.StorableDataSet
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Component
+
+/**
+ * Implementation of a data manager for Dataland including meta data storages
+ */
+@Component("DefaultManager")
+class DataManager(
+    @Autowired var edcClient: DefaultApi
+) : DataManagerInterface {
+    var dataMetaData = mutableMapOf<String, DataMetaInformation>()
+    var companyData = mutableMapOf<String, CompanyMetaInformation>()
+    private var companyCounter = 0
+
+    /*
+    ________________________________
+    Helper-methods:
+    ________________________________
+     */
+
+    private fun verifyCompanyIdExists(companyId: String) {
+        if (!companyData.containsKey(companyId)) {
+            throw IllegalArgumentException("Dataland does not know the company ID $companyId.")
+        }
+    }
+
+    private fun verifyDataIdIsExists(dataId: String) {
+        if (!dataMetaData.containsKey(dataId)) {
+            throw IllegalArgumentException("Dataland does not know the data ID: $dataId.")
+        }
+    }
+
+    private fun verifyDataTypeExists(dataType: String) {
+        val matchesForDataType = dataMetaData.filter { it.value.dataType.equals(dataType) }
+        if (matchesForDataType.isEmpty()) {
+            throw IllegalArgumentException("Dataland does not know the data type: $dataType")
+        }
+    }
+
+    /*
+    ________________________________
+    Methods to route data inserts and queries to the data store and save meta data in the Dataland-Meta-Data-Storage:
+    ________________________________
+     */
+
+    override fun addDataSet(storableDataSet: StorableDataSet): String {
+        verifyCompanyIdExists(storableDataSet.companyId)
+
+        val dataId = edcClient.insertData(storableDataSet.data)
+
+        dataMetaData[dataId] =
+            DataMetaInformation(dataId, dataType = storableDataSet.dataType, companyId = storableDataSet.companyId)
+        companyData[storableDataSet.companyId]!!.dataRegisteredByDataland.add(
+            DataMetaInformation(
+                dataId = dataId,
+                dataType = storableDataSet.dataType,
+                companyId = storableDataSet.companyId
+            )
+        )
+        return dataId
+    }
+
+    override fun getData(dataManagerInputToGetData: DataManagerInputToGetData): String {
+        verifyDataIdIsExists(dataManagerInputToGetData.dataId)
+
+        val data = edcClient.selectDataById(dataManagerInputToGetData.dataId)
+
+        if (data == "") {
+            throw IllegalArgumentException(
+                "No data set with the id: ${dataManagerInputToGetData.dataId} " +
+                    "could be found in the data store."
+            )
+        }
+        if (dataMetaData[dataManagerInputToGetData.dataId]!!.dataType != dataManagerInputToGetData.dataType) {
```

Comment:
Wir sollten diesen Konsistenzcheck machen, bevor wir EuroDaT nerven.

Vielleicht sollte dieser Konsistenzcheck in eine eigene Funktion? (`verifyDataIsOfType(dataId, dataType)` - das könnte dann intern auch `verifyDataIdExists` aufrufen - und dann wäre das einfach der erste Call in der Funktion `getData`

Reply:
fange hiermit jetzt an

Reply:
ok